.github/workflows/release.yml

```diff
@@ -43,26 +44,155 @@ jobs:
           title: Version Packages (v3)
           version: pnpm version-for-release
 
+      - name: Store hasChangesets for subsequent jobs
+        id: store-has-changesets
+        run: |
+          echo "hasChangesets=${{steps.pr.outputs.hasChangesets}}"
+          echo "hasChangesets=${{steps.pr.outputs.hasChangesets}}" >> $GITHUB_OUTPUT
+
+  pack:
+    name: Generate tarballs for publishing
+    needs: pr
+    if: needs.pr.outputs.hasChangesets == 'false'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v5
+
+      - name: Set up the environment
+        uses: ./.github/actions/setup-env
+
+      - name: Install Dependencies
+        run: pnpm install --frozen-lockfile --prefer-offline
```

Comment:
I think `--prefer-offline` could be removed from this workflow. It's a bit faster, because it removes some integrity validations. While I think it's a good trade off for most of our CI workflows, for the release one, i think it isn't.

Reply:
I have removed all of the `--prefer-offline` flags.